tests/sentry/api/endpoints/test_project_ownership.py

```diff
@@ -391,22 +413,53 @@ def test_update_by_member_denied(self):
         assert resp.status_code == 403
 
     def test_turn_off_auto_assignment_clears_autoassignment_cache(self):
-        ProjectOwnership.objects.create(
-            project=self.project, raw="*.js member@localhost #tiger-team"
-        )
         # Turn auto assignment on
         self.client.put(self.path, {"autoAssignment": "Auto Assign to Issue Owner"})
+
+        # Create codeowner rule
+        self.code_mapping = self.create_code_mapping(project=self.project)
+        rule = Rule(Matcher("path", "*.py"), [Owner("team", self.team.slug)])
+        ProjectOwnership.objects.create(
+            project_id=self.project.id, schema=dump_schema([rule]), fallthrough=True
+        )
+        self.create_codeowners(
+            self.project, self.code_mapping, raw="*.py @tiger-team", schema=dump_schema([rule])
+        )
+
+        # Auto assign rule using codeowner
+        self.event = self.store_event(
+            data=self.python_event_data(),
+            project_id=self.project.id,
+        )
+        GroupOwner.objects.create(
+            group=self.event.group,
+            type=GroupOwnerType.CODEOWNERS.value,
+            user_id=None,
+            team_id=self.team.id,
+            project=self.project,
+            organization=self.project.organization,
+            context={"rule": str(rule)},
+        )
+        ProjectOwnership.handle_auto_assignment(self.project.id, self.event)
+
         auto_assignment_ownership = ProjectOwnership.objects.get(project=self.project)
         auto_assignment_types = ProjectOwnership._get_autoassignment_types(
             auto_assignment_ownership
         )
+        assert auto_assignment_types == [
+            GroupOwnerType.OWNERSHIP_RULE.value,
+            GroupOwnerType.CODEOWNERS.value,
+        ]
         # Get the cache keys
         groups = Group.objects.filter(
             project_id=self.project.id,
             last_seen__gte=timezone.now() - timedelta(seconds=ISSUE_OWNERS_DEBOUNCE_DURATION),
-        ).values_list("id", flat=True)
+        )
+        assert groups is not None
```

Comment:
this should probably be `assert groups` (since it'll always be non-None -- but sometimes it might be empty)

Reply:
good catch, thank you!